sympy/tensor/array/ndim_array.py

```diff
@@ -70,10 +70,12 @@ def __new__(cls, iterable, shape=None, **kwargs):
 
     def _parse_index(self, index):
 
-        if isinstance(index, (SYMPY_INTS, Integer)):
-            if index >= self._loop_size:
-                raise ValueError("index out of range")
-            return index
+#        if isinstance(index, (SYMPY_INTS, Integer)):
```

Comment:
Maybe you still need to handle this case. What about raising an exception?

Reply:
Yes, I agree with you. We need to avoid passing an Integer to `_parse_index`, only the tuple index should be accepted. Raising an exception is more informative.